genestack_client/genestack_connection.py

```diff
@@ -180,15 +186,56 @@ def open(self, path, data=None, follow=True, headers=None):
         if headers:
             _headers.update(headers)
         try:
-            response = self.session.post(self.server_url + path, data=data, headers=_headers,
+            url = self.server_url + path
+            response = self.session.post(url, data=data, headers=_headers,
                                          allow_redirects=follow, stream=True)
             if response.status_code == 401:
                 raise GenestackAuthenticationException('Authentication failure')
             try:
                 response.raise_for_status()
-                return response.raw
+                return urllib.addinfourl(StringIO(response.raw.read(decode_content=True)),
+                                         headers, url, code=response.status_code)
+            except HTTPError as e:
+                raise GenestackResponseError(*e.args)
+        except RequestException as e:
+            raise GenestackConnectionFailure(str(e))
+
+    def get_response(self, path, data=None, follow=True, headers=None):
+        """
+        Get response from an URL. The URL is the concatenation of the server URL and ``path``.
+
+        :param path: part of URL that is added to self.server_url
+        :param data: dict of parameters, file-like objects or strings
+        :param follow: should we follow a redirection if any?
+        :param headers: dictionary of additional headers; list of pairs is
+                        supported too until v1.0 (for backward compatibility)
+        :type headers: dict[str, str] | list[tuple[str]]
+        :return: response
+        :rtype: Response
+        """
+        if data is None:
+            data = ''
+        elif isinstance(data, dict):
+            data = urllib.urlencode(data)
```

Comment:
> :param data: dict of parameters, file-like objects or strings

Doesn't look like anything except strings and dict / list of tuples is supported.

Reply:
Actually it is:
Dictionary, bytes, or file-like object to send in the body of the :class:`Request`.

Looks like we should remove encoding